hbase-mapreduce/src/test/java/org/apache/hadoop/hbase/mapreduce/TestWALRecordReader.java

```diff
@@ -335,13 +342,16 @@ private void testSplitWithMovingWAL(InputSplit split, byte[] col1, byte[] col2)
     // Move log file to archive directory
     // While WAL record reader is open
     WALInputFormat.WALSplit split_ = (WALInputFormat.WALSplit) split;
-
     Path logFile = new Path(split_.getLogFileName());
-    Path archivedLog = AbstractFSWALProvider.getArchivedLogPath(logFile, conf);
-    boolean result = fs.rename(logFile, archivedLog);
-    assertTrue(result);
-    result = fs.exists(archivedLog);
-    assertTrue(result);
+    Path archivedLogDir = getWALArchiveDir(conf);
+    Path archivedLogLocation = new Path(archivedLogDir, logFile.getName());
+    assertNotEquals(split_.getLogFileName(), archivedLogLocation.toString());
+
+    assertTrue(fs.rename(logFile, archivedLogLocation));
+    assertTrue(fs.exists(archivedLogDir));
+    assertFalse(fs.exists(logFile));
+    // TODO: This is not behaving as expected. WALInputFormat#WALKeyRecordReader doesn't open
```

Comment:
This test is not testing what it is intending to.
Before this patch, it was using` AbstractFSWALProvider.getArchivedLogPath(logFile, conf);` to get `archivedLogLocation`. But `AbstractFSWALProvider#getArchivedLogPath` was returning the same path as `logFile`  from [here](https://github.com/apache/hbase/blob/master/hbase-server/src/main/java/org/apache/hadoop/hbase/wal/AbstractFSWALProvider.java#L498)

After this patch, we are able to successfully rename the file to oldWALs directory but somehow it is not triggering the condition within [nextKeyValue method](https://github.com/apache/hbase/blob/d56db9d1b240626ecbfea7ba98658befcc162afe/hbase-mapreduce/src/main/java/org/apache/hadoop/hbase/mapreduce/WALInputFormat.java#L223) to look into archiveDir.

Reply:
So you mean we should file another issue to address this? Anyway, I think this PR is a critical one as it could cause data loss, so let me merge it to all branches first in case of not blocking the upcoming 2.4.x release.

Reply:
I was going to cut 2.4.6RC0 today but saw this issue. Will wait until this is merged to branch-2.4.

Reply:
>  So you mean we should file another issue to address this

Yes, we need to open a new jira to modify the test.

Reply:
Created https://issues.apache.org/jira/browse/HBASE-26247